ui/app/components/app/whats-new-popup/whats-new-popup.js

```diff
@@ -0,0 +1,174 @@
+import React, { useContext, useMemo, useRef, useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { I18nContext } from '../../../contexts/i18n';
+import { useEqualityCheck } from '../../../hooks/useEqualityCheck';
+import Button from '../../ui/button';
+import Popover from '../../ui/popover';
+import { updateViewedNotifications } from '../../../store/actions';
+import { getTranslatedUINoficiations } from '../../../../../shared/notifications';
+import { getSortedNotificationsToShow } from '../../../selectors';
+
+function getActionFunctions() {
+  const actionFunctions = {
+    1: () => {
+      global.platform.openTab({
+        url: 'https://metamask.io/download.html',
+      });
+    },
+    2: () => {
+      global.platform.openTab({
+        url:
+          'https://survey.alchemer.com/s3/6173069/MetaMask-Extension-NPS-January-2021',
+      });
+    },
+    3: () => {
+      global.platform.openTab({
+        url: 'https://community.metamask.io/t/about-the-security-category/72',
+      });
+    },
+  };
+
+  return actionFunctions;
+}
+
+export default function WhatsNewPopup({ onClose }) {
+  const t = useContext(I18nContext);
+
+  const actionFunctions = getActionFunctions();
+
+  const notifications = useSelector(getSortedNotificationsToShow);
+
+  const containerRef = useRef();
+
+  const memoizedNotifications = useEqualityCheck(notifications);
+  const idRefMap = useMemo(
+    () =>
+      memoizedNotifications.reduce(
+        (_idRefMap, notification) => ({
+          ..._idRefMap,
+          [notification.id]: React.createRef(),
+        }),
+        {},
+      ),
+    [memoizedNotifications],
+  );
+
+  const [seenNotifications, setSeenNotifications] = useState({});
+  const observationsCreated = useRef(false);
+
+  useEffect(() => {
+    const observers = [];
+    observationsCreated.current = true;
+
+    Object.entries(idRefMap).forEach(([id, ref]) => {
+      const observer = new window.IntersectionObserver(
+        ([entry]) => {
+          if (entry.isIntersecting) {
+            setSeenNotifications((_seenNotifications) => ({
```

Comment:
I'd suggest [_`unobserving`_](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver/unobserve) each element here to avoid further callbacks as the user scrolls back and forth.

Reply:
Done in 2d7b8709a